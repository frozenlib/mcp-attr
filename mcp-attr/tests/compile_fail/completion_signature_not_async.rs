use mcp_attr::{server::{mcp_server, McpServer, RequestContext, complete_fn}, Result};

struct TestServer;

#[complete_fn]
fn complete_not_async(_value: &str, _cx: &RequestContext) -> Result<Vec<String>> {
    Ok(vec!["test".to_string()])
}

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(complete_not_async)] msg: String) -> Result<String> {
        Ok(msg)
    }
}

fn main() {}