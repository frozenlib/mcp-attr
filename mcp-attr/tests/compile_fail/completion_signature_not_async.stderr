error: completion function must be async
 --> tests/compile_fail/completion_signature_not_async.rs:6:4
  |
6 | fn complete_not_async(_value: &str, _cx: &RequestContext) -> Result<Vec<String>> {
  |    ^^^^^^^^^^^^^^^^^^

warning: unused import: `RequestContext`
 --> tests/compile_fail/completion_signature_not_async.rs:1:48
  |
1 | use mcp_attr::{server::{mcp_server, McpServer, RequestContext, complete_fn}, Result};
  |                                                ^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0425]: cannot find function `complete_not_async` in this scope
  --> tests/compile_fail/completion_signature_not_async.rs:13:44
   |
13 |     async fn test_prompt(&self, #[complete(complete_not_async)] msg: String) -> Result<String> {
   |                                            ^^^^^^^^^^^^^^^^^^ not found in this scope
